compiler/src/dotty/tools/dotc/core/SymDenotations.scala

```diff
@@ -1564,14 +1564,14 @@ object SymDenotations {
     initPrivateWithin: Symbol)
     extends SymDenotation(symbol, maybeOwner, name, initFlags, initInfo, initPrivateWithin) {
 
-    import util.HashTable
+    import util.IdentityHashMap
```

Comment:
`util.IdentityHashMap` and `java.util.IdentityHashMap` are too easy to confuse IMHO. Can we use a different name like `EqHashMap`?

Reply:
In fact, if you happen to write your package declaration as `package dotty.tools.dotc.core`, then `import util.IdentityHashMap` will compile but import the Java map!

Reply:
That's somewhat intentional. There's no reason anymore to use a java.util.IdentityHashMap. The dotc.util map is uniformly better in both performance and types.

Reply:
> In fact, if you happen to write your package declaration as package dotty.tools.dotc.core, then import util.IdentityHashMap will compile but import the Java map!

Ah I see what you mean now. Yes, we should do something about that.